src/text/non_blocking/raw_text_reader.rs

```diff
@@ -378,9 +383,12 @@ impl<T: ToIonDataSource> RawTextReader<T> {
         // EOF are ambiguous.
         const SENTINEL_ION_TEXT: &str = "\n0\n";
         // Make a note of the buffer's length; we're about to modify it.
-        let original_length = self.buffer.remaining_text().len();
+        let mut remaining_text = self.buffer.remaining_text().to_owned();
```

Comment:
Since this only happens at EOF, can we not `append_bytes()` to the buffer rather than allocating a new `String`?

Reply:
Ah, of course! That makes sense, thanks for explaining. I wonder if we can optimize this another way, but solving that puzzle can wait. :)